File: funcscript/parser/syntax/FuncScriptParser.GetCommentBlock.cs
namespace funcscript.core
{
    public partial class FuncScriptParser
    {
        record GetCommentBlockResult(ParseNode ParseNode, int NextIndex)
            :ParseResult(ParseNode,NextIndex);
        static GetCommentBlockResult GetCommentBlock(ParseContext context, int index)
        {
            ParseNode parseNode = null;
            var i = GetLiteralMatch(context, index, new string[] { "//" }).NextIndex;
            if (i == index)
                return new GetCommentBlockResult(parseNode, index);
            var i2 = context.Expression.IndexOf("\n", i,StringComparison.InvariantCulture);
            if (i2 == -1)
                i = context.Expression.Length;
            else
                i = i2 + 1;
            parseNode = new ParseNode(ParseNodeType.Comment, index, i - index);
            return new GetCommentBlockResult(parseNode, i);
        }
    }
}

File: ts/core/src/parser/syntax/FuncScriptParser.GetCommentBlock.ts
import { ParseContext, ParseNode, ParseNodeType, ParseResult } from "../FuncScriptParser.Main";  
import { GetLiteralMatchMultiple } from "./FuncScriptParser.GetLiteralMatch";

export interface GetCommentBlockResult extends ParseResult {}  

export function GetCommentBlock(context: ParseContext, index: number): GetCommentBlockResult {  
    let parseNode: ParseNode | null = null;  
    let i = GetLiteralMatchMultiple(context, index, ["//"]).NextIndex;  

    if (i === index) {  
        return { ParseNode: parseNode, NextIndex: index };  
    }  

    const i2 = context.Expression.indexOf("\n", i);  

    if (i2 === -1) {  
        i = context.Expression.length;  
    } else {  
        i = i2 + 1;  
    }  

    parseNode = new ParseNode(ParseNodeType.Comment, index, i - index);  

    return { ParseNode: parseNode, NextIndex: i };  
}  